pcpp/parser.py

```diff
@@ -111,15 +111,12 @@ def t_CPP_LINECONT(t):
 # Identifier
 t_CPP_ID = r'[A-Za-z_][\w_]*'
 
-# Integer literal
-def CPP_INTEGER(t):
-    r'(((((0x)|(0X))[0-9a-fA-F]+)|(\d+))([uU][lL]|[lL][uU]|[uU]|[lL])?)'
+# Preprocessor number
+def PP_NUMBER(t):
+    r'\.?\d(\.|[\w_]|\'[\w_]|[eEpP][-+])*'
```

Comment:
These suggestions are all committed

1. The regex is indeed waaaay lax, but only because the pp-number specification itself is lax  
    (my initial PR message links this proposal to tighten it up https://wg21.link/D2180R0)
2. The `\'` was only needed because the string was `'...'` quoted, so I switched to `"..."`
3. The capture group is gone (forgot to mention that in the commit message)

Your regex comment in the main ticket also raises the unicode question;
how to add XID_Continue to identifier-continue? I'm following up on this and will comment when enlightened.

(Your #71 proposal to tighten up the `CPP_INTEGER` token is superseded by this PR as it entirely removes the token - I'll discuss that further in another comment)